### PR TITLE
chore: support report in vexination walker

### PR DESCRIPTION
**Pull request by bxf12315**

### CODE REVIEW

**Review by ctron — APPROVED** [review]

**Comment by jcrossley3** [comment]
Some background on this: https://issues.redhat.com/browse/TC-910

File: vexination/walker/src/server.rs [hunk 1]
```diff
@@ -42,103 +54,166 @@ pub async fn run(
...
+                                        .post(sink.clone())
+                                        .header(header::CONTENT_TYPE, "application/json")
+                                        .body(b.clone())
+                                        .inject_token(&provider)
+                                        .await?
+                                        .send()
+                                        .await
+                                    {
+                                        Ok(r) if r.status() == StatusCode::CREATED => {
+                                            log::info!("VEX ({}) stored successfully", &adv.csaf.document.tracking.id);
```
Comment (bobmcwhirter) [inline vexination/walker/src/server.rs:104]: Probably should break out of the loop upon success, yeah?

Comment (ctron) [reply vexination/walker/src/server.rs:104]: Yes!

Comment (jcrossley3) [reply vexination/walker/src/server.rs:104]: Can we use a `SendVisitor` instead? To better encapsulate the retry behavior. Perhaps incorporate these changes within #1097 ?

**Pull request merged**